sdk/servicebus/service-bus/src/receivers/sessionReceiver.ts

```diff
@@ -139,51 +157,43 @@ export class SessionReceiverImpl<ReceivedMessageT extends ReceivedMessage | Rece
     sessionOptions: CreateSessionReceiverOptions,
     retryOptions: RetryOptions = {}
   ): Promise<SessionReceiver<ReceivedMessageT>> {
+    context.isSessionEnabled = true;
+    if (sessionOptions.sessionId != undefined) {
+      sessionOptions.sessionId = String(sessionOptions.sessionId);
+    }
+    const messageSession = await MessageSession.create(context, {
+      sessionId: sessionOptions.sessionId,
+      autoRenewLockDurationInMs: sessionOptions.autoRenewLockDurationInMs,
+      receiveMode: convertToInternalReceiveMode(receiveMode)
+    });
     const sessionReceiver = new SessionReceiverImpl<ReceivedMessageT>(
+      messageSession,
       context,
       receiveMode,
       sessionOptions,
       retryOptions
     );
-
-    await sessionReceiver._createMessageSessionIfDoesntExist();
     return sessionReceiver;
   }
 
   private _throwIfReceiverOrConnectionClosed(): void {
     throwErrorIfConnectionClosed(this._context.namespace);
     if (this.isClosed) {
-      const errorMessage = getReceiverClosedErrorMsg(
-        this._context.entityPath,
-        this._context.isClosed,
-        this.sessionId!
-      );
+      const errorMessage = getReceiverClosedErrorMsg(this._context.entityPath, this.sessionId!);
       const error = new Error(errorMessage);
       log.error(`[${this._context.namespace.connectionId}] %O`, error);
       throw error;
     }
   }
 
-  private async _createMessageSessionIfDoesntExist(): Promise<void> {
-    // TODO - pass timeout for MessageSession creation
-    if (this._messageSession) {
-      return;
+  private _throwIfMessageSessionDoesntExist(): void {
```

Comment:
Since this is used only for receiving messages, can we move it into `_throwIfAlreadyReceiving()`?

Reply:
Can merge both.

Reply:
On second thoughts..., given that `_throwIfReceiverOrConnectionClosed()` checks if the message session exists in the context cache, do we even need this check?
If the only difference is checking `isOpen()`, then can we not move this change to `_throwIfReceiverOrConnectionClosed()`

Reply:
On a related note, `isClosed` should be updated from

`this._isClosed || (this.sessionId ? !this._context.messageSessions[this.sessionId] : false)`

to 

`this._isClosed || this._context.messageSessions[this.sessionId]`

Reply:
_throwIfReceiverOrConnectionClosed should be updated to
- Throw if connection is closed 
- Throw if user closed the session receiver
- Throw if the internal `messageSession` got closed and so is not available in the context cache
- Throw if the internal `messageSession` exists in cache, but `isOpen()` returns false

The first two cases would retain the error message from before
The last two cases should use the SessionLockLostError

Reply:
Updated as suggested.
```ts
  private _throwIfReceiverOrConnectionClosed(): void {
    throwErrorIfConnectionClosed(this._context.namespace);
    if (this._isClosed) {
      const errorMessage = getReceiverClosedErrorMsg(this._context.entityPath, this.sessionId!);
      const error = new Error(errorMessage);
      log.error(`[${this._context.namespace.connectionId}] %O`, error);
      throw error;
    }
    if (!this._context.messageSessions[this.sessionId] || !this._messageSession.isOpen()) {
      const amqpError: AmqpError = {
        condition: ErrorNameConditionMapper.SessionLockLostError,
        description: `The session lock has expired on the session with id ${this.sessionId}`
      };
      throw translate(amqpError);
    }
  }
```

Reply:
Related... Should the `isClosed` be updated from

`this._isClosed || !this._context.messageSessions[this.sessionId]`

to 

`this._isClosed || !this._context.messageSessions[this.sessionId] || !this._messageSession.isOpen()`

Reply:
Agreed. Especially because we don't create a new message session on behalf of the user if the current one is closed under the hood due to any error

Reply:
Updated